src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,205 @@
+    motorInit(m_leftMotor2, DriveConstants.kLeftEncoderInverted);
+
+
+    addChild("Left Motors", m_leftMotors);
+    addChild("Right Motors", m_rightMotors);
+    addChild("Drivetrain", m_Drive);
+
+  }
+
+private void encoderInit(CANEncoder encoder) {
+    // Converts the input into desired value for distance and velocity
+    encoder.setPositionConversionFactor(DriveConstants.kEncoderDistancePerPulse); // ADD kEncoderDistancePerPulse TO CONSTANTS
+    encoder.setVelocityConversionFactor(DriveConstants.kEncoderSpeedPerPulse); // ADD kEncoderSpeedPerPulse TO CONSTANTS
+    encoderReset(encoder);
+  }
+
+private void encoderReset(CANEncoder encoder) {
+    encoder.setPosition(0);
+  }
+
+public void resetLeftEncoders() {
+    // Reset all left-side drive encoders
+    encoderReset(m_leftMotorEncoder1);
+    encoderReset(m_leftMotorEncoder2);
+  }
+
+public void resetRightEncoders() {
+    // Reset all right-side drive encoders
+    encoderReset(m_rightMotorEncoder1);
+    encoderReset(m_rightMotorEncoder2);
+  }
+
+public void resetAllEncoders() {
+    // Reset all drive encoders
+    resetLeftEncoders();
+    resetRightEncoders();
+  }
+
+private void motorInit(CANSparkMax motor, boolean invert) {
+    motor.restoreFactoryDefaults(); // Reset settings in motor in case they are changed
+    motor.setIdleMode(IdleMode.kBrake); // Sets the motors to brake mode from the beginning
+
+    encoderInit(motor.getEncoder()); // Initializes encoder within motor
+  }
+
+// Gets left side distance by averaging left encoders
+private double getLeftEncoderAverage() {
+    return (m_leftMotorEncoder1.getPosition() + m_leftMotorEncoder2.getPosition()) / 2.0;
+  }
+
+// Gets right side distance by averaging right encoders
+private double getRightEncoderAverage() {
+    return (m_rightMotorEncoder1.getPosition() + m_rightMotorEncoder2.getPosition()) / 2.0;
+  }
+
+// Gets average distance by averaging the right and left distance
+public double getDistance() {
+    return (getRightEncoderAverage() + getLeftEncoderAverage()) / 2.0;
+  }
+
+// Drives the motor using tank drive
+public void tankDrive(double leftPower, double rightPower) {
+      m_Drive.tankDrive(leftPower, rightPower);
+    }
+  
+// Drives the motors using arcade drive
+public void arcadeDrive(double speed, double turn) {
+      m_Drive.arcadeDrive(speed, turn, true);
+    }
+  
+// Drives the motor using trigger drive
+public void triggerDrive(double reverse, double forward, double turn) {
+      arcadeDrive(forward - reverse, turn);
+    }
+  
+// sets motor values to zero to stop
+public void stopDrive() {
+      m_Drive.tankDrive(0,0);
+    }
+
+// all motors go at half speed when slowSpeed is activated 
+public void slowSpeed() {
+      m_Drive.setMaxOutput(0.5);
+      slow = true;
+    }
+  
+// all motors go at full speed when regularSpeed is activated 
+public void regularSpeed() {
+      m_Drive.setMaxOutput(1.0);
+      slow = false;
+    }
```

Comment:
I would change these to gear shifting. public void shiftLow() and public void shiftHigh(), maybe?